---
title: "Problemset 2"
author: "Malay Basu (malay@uab.edu)"
output: pdf_document
---

# Problem 1
You can call `data(airquality)` in R. It will generate the following data.
```{r}
data("airquality")
knitr::kable(head(airquality))

```
(a) Calculate the pairwise Pearson correlation of all the variables of this dataset and create a dataframe that has columns like this.

| var1 | var2 | corr|
|------|------|-----|
|Ozone |Solar | 0.4 |

```{r}
head(airquality)

correlation_matrix <- cor(airquality, method = "pearson", use = "pairwise.complete.obs")

library(Hmisc)

format_df_test <- function(cormat) {
  ut <- upper.tri(cormat)
  data.frame(
    var1 = rownames(cormat)[row(cormat)[ut]], 
    var2 = rownames(cormat)[col(cormat)[ut]], 
    corr = (cormat)[ut]
  )
}

formated_table <- format_df_test(correlation_matrix)
formated_table
knitr::kable(formated_table, digits = 2)

```


(b) Using `ggplot` draw the scatterplot of the variables that show the highest correlation. You can arbitrarily choose one of the two variables as independent. Make the scatterplot publication quality. Also calculate the r.sq of the plot and put it on the top of the plot as subtitle of the plot. 
```{r}

library(ggplot2)
#install.packages("ggpmisc")
library(ggpmisc)

my.formula <- y ~ x
ggplot(data = airquality, aes(x = Ozone, y = Temp)) +
  geom_smooth(method = "lm", se=FALSE, formula = my.formula) +
  stat_poly_eq(formula = my.formula, 
               aes(label = paste(..rr.label.., sep = "~~~")), 
               parse = TRUE) +
  geom_point() +
  ggtitle("Temperature Generally Increases with Ozone") +
  theme(plot.title = element_text(hjust = 0.5)) +
  labs(y = "Temperature")

```



# Problem 2
Write an R script that takes two arguments: (1) a fasta file name, (2) a sequence ID. The script should print out the sequence matching the id in FASTA format to the terminal.

```{r}


#BiocManager::install("Biostrings")
library(Biostrings)
library(stringr)

###write fild containing homologene ID of gene##
##Load homologene data
#ids <- "~/CB2-101/test_class/assignment2/homologene.data"

##read homologene data as a table
#ids <- read.table(ids, stringsAsFactors = F, header = F, fill = T)

##Search for your ID of interest V1=HomologeneID
#ids <- ids[ids$V1 == "460",]

##pull out the IDs you want (row 1 is human)
#ids <- ids$V6[c(1)]

##write the ID(s) of interest to a table
#write.table(ids, "human_tp53.ids", quote = F, row.names = F, col.names = F)

##fasta arg is fasta file of choice

##save output file as .fas

args <- commandArgs(trailingOnly = T)
fasta <- args[1]
ids <- args[2]
outfile <- args[3]

# Read fasta and extrat ids
seqs <- readAAStringSet(fasta)
idset <- names(seqs)

pattern <- "^(\\S+)\\."
matches <- str_match(idset, pattern)
idset <- matches[,2]

# Read required id files
required_ids <- read.table(ids, stringsAsFactors = F, header = F, fill = T)
matches <- str_match(required_ids$V1, pattern = pattern)
required_ids <- matches[,2]

# Which ids matches
matching_index <- which(idset %in% required_ids)

# Subset the fasta files
seq_subset <- seqs[matching_index]

# Write out the subset as fasta
writeXStringSet(seq_subset, outfile)

#script saved as question2.R

```
```{bash}
Rscript question2.R p53.fas human_tp53.ids output_question_2.fas

cat output_question_2.fas
```




# Problem 3

Using `wget` download BLOSUM62 matrix from NCBI FTP server (ftp://ftp.ncbi.nih.gov/blast/matrices/BLOSUM62). Process it in anyway you can and read it in R as a matrix, a dataframe, or a list. You should store the data such a way that you can call the score given two amino acids as key as a fast lookup table. Read the accompanied `ex_align.fas` file and calculate the score of the given alignment. Consider each indel has score 0. The alignment file is in aligned fasta format.

Tips: You need to use either `seqnir` or `Biostrings` package and loop through each position in the alignment. 


```{bash}
curl -O ftp://ftp.ncbi.nih.gov/blast/matrices/BLOSUM62
```


```{r}
library(Biostrings)
#install.packages("seqinr")
library(seqinr)

data("BLOSUM62")
str(BLOSUM62)


test <- readAAStringSet("~/CB2-101/test_class/assignment2/ex_align.fas")

pattern <- AAString(test$P1)
subject <- AAString(test$P2)

pal <- pairwiseAlignment(pattern, subject, substitutionMatrix="BLOSUM62", gapOpening=0, gapExtension=0)


```
















