---
title: "Problem_Set_2_CGW"
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## GitHub Documents

This is an R Markdown format used for publishing markdown documents to GitHub. When you click the **Knit** button all R code chunks are run and a markdown file (.md) suitable for publishing to GitHub is generated.



# Assignment 2 Problem 1

You can call `data(airquality)` in R. It will generate the following data.
```{r}
data("airquality")
knitr::kable(head(airquality))
```

# Problem1_a
Calculate the pairwise Pearson correlation of all the variables of this dataset and create a dataframe that has columns like this.

| var1 | var2 | corr|
|------|------|-----|
|Ozone |Solar | 0.4 |

```{r}


airquality.cor = cor(airquality, 
                     use="complete.obs",
                     method = "pearson"
)

```

airquality.cor presents all the pearson correlation variance values, but in a different format than asked for in the question. To create a dataframe that has columns like those in the question:
```{r}

library(dplyr)
stats <- data.frame(var1 = "var1", var2="var2", corr=1)
stats <- stats%>% mutate(var1 = as.character(var1),
                           var2 = as.character (var2),
                           corr = as.numeric(corr))
head(stats)
```
```{r}

for(i in 1:6) {
  for(j in 1:6) {
    a <- cor.test(airquality[,i],airquality[,j])
    row <- c(i,j, a$estimate)
    stats <- rbind(stats, row)
  }
  }
head (stats)
stats <- stats %>% filter (corr !=1)
```
```{r}

stats[stats == "1"] <- "Ozone"
stats[stats == "2"] <- "Solar.R"
stats[stats == "3"] <- "Wind"
stats[stats == "4"] <- "Temp"
stats[stats == "5"] <- "Month"
stats[stats == "6"] <- "Day"
View(stats)
```

# Problem1_b
Using `ggplot` draw the scatterplot of the variables that show the highest correlation. You can arbitrarily choose one of the two variables as independent. Make the scatterplot publication quality. Also calculate the r.sq of the plot and put it on the top of the plot as subtitle of the plot. 
```{r}

b <- lm(formula=airquality$Temp~airquality$Ozone)
summary (b)
#Adjusted R-squared value: 0.4832

library(ggplot2)
ggplot(airquality, aes(x=Temp, y=Ozone)) +
  geom_point ()+
  ggtitle("Correlation between Ozone and Temperature", subtitle="R^2=0.4832") +
  geom_smooth(method = "lm", se=F, color="black") +
  theme_bw() +
  xlab("Temperature") +
  ylab("Ozone") +
  theme(plot.title = element_text(hjust = 0.5),
        plot.subtitle = element_text(hjust=0.5))
```

# Assignment 2 Problem 2
Write an R script that takes two arguments: (1) a fasta file name, (2) a sequence ID. The script should print out the sequence matching the id in FASTA format to the terminal.



```{r}
Prob_2_answer <- function(a, b) {
  
library("Biostrings")
fastaFile = readDNAStringSet('sample.fa')
seq_name = names(fastaFile)
sequence = paste(fastaFile)
df <- data.frame(seq_name, sequence)

g <- grep(b, seq_name)
df[g,2]
}

```



# Assignment 2_Problem 3

Using `wget` download BLOSUM62 matrix from NCBI FTP server (ftp://ftp.ncbi.nih.gov/blast/matrices/BLOSUM62). Process it in anyway you can and read it in R as a matrix, a dataframe, or a list. You should store the data in such a way that you can call the score given two amino acids as key as a fast lookup table. Read the accompanied `ex_align.fas` file and calculate the score of the given alignment. Consider each indel has score 0. The alignment file is in aligned fasta format.

Tips: You need to use either `seqnir` or `Biostrings` package and loop through each position in the alignment. 
```{r}
#Problem #2

#Load BLOSUM62 & writing it as a list


Prob_2_answer <- function(a) {
#data("BLOSUM62")
#write.table(BLOSUM62, "BLOSUM62.txt" ,quote=F)

apples <- read.table("BLOSUM62", header=T)

apples2.0 <-data.frame(A=0, R=0, N=0, D=0, C=0, Q=0, E=0,
                       G=0, H=0, I=0, L=0, K=0, M=0, F=0, P=0, S=0, T=0, W=0, 
                       Y=0, V=0, B=0, Z=0, X=0, X.=0)
row.names(apples2.0) <- "-"
apples3.0 <- rbind(apples, apples2.0)

apples4.0 <- data.frame("k"=rep(0,25))
colnames(apples4.0) <- "-"            
apples4.0
cbind(apples3.0, apples4.0)


# x <-"A"
# apples[x,x] + apples["A","Q"]

a <- "ex_align.fas"

# Making the input file usable
library("Biostrings")
fastaFile = readAAStringSet(a)
seq_name = names(fastaFile)
sequence = paste(fastaFile)
df <- data.frame(seq_name, sequence)

b <- df[1,2]
b1 <-as.character(b)
b2 <-strsplit(b1, "")
b3 <- unlist(b2)

c <- df[2,2]
c1 <-as.character(c)
c2 <-strsplit(c1, "")

c3 <-unlist(c2)

test_input <-rbind(b3,c3)
ncol(test_input)
}
nrow(test_input)
test_input[1,1]

yup <- rep(0,107)

for (i in 1:107){
  
yup[i] <- apples4.0[test_input[1,i],test_input[2,1]]

}

yup

for (i in 1:107){
  cat(apples4.0[test_input[1,i],test_input[2,i]])
} 


```


```{r}
